src/go/k8s/pkg/resources/configmap.go

```diff
@@ -352,7 +352,7 @@ func (r *ConfigMapResource) CreateConfiguration(
 	}
 
 	cfg.SetAdditionalRedpandaProperty("auto_create_topics_enabled", r.pandaCluster.Spec.Configuration.AutoCreateTopics)
-	cfg.SetAdditionalRedpandaProperty("enable_idempotence", false)
+	cfg.SetAdditionalRedpandaProperty("enable_idempotence", true)
```

Comment:
Should it be conditional on the redpanda version? How does v21 behaves with this enabled?

Reply:
Good question. I will guard it with feature flag package.